baybe/parameters/categorical.py

```diff
@@ -24,7 +24,7 @@ class CategoricalParameter(DiscreteParameter):
 
     # object variables
     _values: tuple[str, ...] = field(
-        converter=tuple,
+        converter=lambda x: tuple(sorted(x)),
```

Comment:
I still get different results when providing the parameters themselves in a different order: I just included the following lines to permute the order of the parameters, and every result is different:
<img width="664" alt="image" src="https://github.com/user-attachments/assets/976c17aa-a666-448d-a84d-92e799fc2551">
which yields
<img width="448" alt="image" src="https://github.com/user-attachments/assets/0965c94c-d503-4dc2-ae0a-94844d48c6de">
<img width="450" alt="image" src="https://github.com/user-attachments/assets/0de83b97-9617-4307-aa11-e27e164713b6">
<img width="448" alt="image" src="https://github.com/user-attachments/assets/97634b6d-aa73-4df4-ab9a-12269eb4b418">

Reply:
Hi @AVHopp, thanks for raising the issue. In fact, when drafting the PR, I didn't have this kind of permutation in mind. 

Anyway, now that you made me aware of the situation, I spent a while looking into it. At the moment, I'm not sure if we should actually fix this, tbh. While it *could* be fixed, the necessary changes are quite invasive and have consequences other places as well. The problem is that, in order to fix it, we'd need to ensure that the `exp_rep` dataframe remains permutation-invariant, regardless of the user input and in particular the constructor being used. So the necessary changes and side-effects that I currently see are:
* We'd need to define a set of rules for ordering parameters (have played around and found a good solution)
* We'd need to order the `parameters` attribute in both subspaces (<-- easy fix)
* We'd need to sort to the incoming `exp_rep` argument (e.g. using a converter) in the `__init__` function. 

The latter is also doable but
* Not sure if too costly (<-- perhaps we can ignore this)
* The instantiated dataframe will then obviously be different comparted to what is provided by the user. This effect would be most noticable when using the `from_dataframe` constructor, where the stored `exp_rep` would look very different from what the user specified. Essentially, we run into the same situation with the parameter themselves, where we also sort the user-provided values before we store them. But a potential problem I could see with the searchspace dataframe is when people rely on the order of the frame they've passed, e.g. when they modify the searchspace `metadata`, silently assuming it comes in the same order.

The latter problem would arise if we took a different approach to metadata handling, which is one of the things I've on my list since the early days. But this is definitely further down the road.

@AVHopp, @Scienfitz, thoughts?

Reply:
It seems that there is no easy "yeah we can quickly do this without too much trouble" solution, and I am not sure how much trouble sorting the `exp_rep` will actually be for the user. Most importantly, I think this is too much of a change for this PR, and if we decide to include this, it should have a separate PR (whether or not we want to do that, maybe with giving the user the option to keep the original order is another question). So I'd keep this as it is here.

Only question is: Do we want to mention that the performance of the optimization might depend on the order of parameters somewhere/investigate this effect?

Reply:
perhaps you're overcomplicating things, I think the parameter order just has to be maintained by constructors that use sequences of parameters

- simply order those by parameter name
- do not touch what happens with constructors that provide the exp rep directly

I'm accepting this PR already now because imo the invariance to the order of the parameter objects is optional (the value invariance imo was mroe important), and I would only advocate for it if you're up to have it in the above propose simple manner, the complicated one imo is fully overkill and should not stop this PR

Reply:
Ok, in that case I'll leave the decision up to @AdrianSosic , but just keeping everything as it is is then perfectly fine for me, and I agree that we should only implement the simple solution (if at all).

Reply:
@Scienfitz: I also thought about your proposed solution before you mentioned it but I saw a little downside in that you need to ensure the ordering within each constructor and you cannot handle it in one central place. That also means when we add a new construction, there are chances we forget it. So what do you prefer: add it or leave it?

Reply:
unless you expect one new constructor per week Im not worried by that and it wouldnt even be a massive issue if it was forgotten for a new one

Reply:
@Scienfitz: what do you think about the case where people pass a dataframe? You suggested that we shouldn't touch the dataframe, but what about the corresponding parameters being stored? Should we keep them in the same order as defined via the dataframe or still order them (because these are really two independent things)

Reply:
I've now changed it such that we always order the parameter attributes, which has the advantage that we can do it in the constructor even. Is there anything that speaks against this approach? If not, we can basically merge

Reply:
sounds good lets go